### PR TITLE
sg: add flag to rfc to determine public or private search

### PR DESCRIPTION
**Pull request by burmudar**
By using Corpura "drive" and specifying a `driveId` we were only searching one RFC folder even though in our query we specified that parents should be from two folders. To search across drives you have to use Copura "allDrives", while this gets us all RFCs it makes it hard to select RFCs with the same number like `RFC 748`.

This PR adds a flag to RFC which allows you to select when you want to search private RFCs by specifying the `--private` flag, otherwise it is assumed you want to search public RFCs

With private flag
```
go run ./dev/sg rfc --private list
...
...
...
RFC 742: *******
RFC 743: *******
RFC 748 REVIEW *******
RFC PRIVATE 750: *******
```
Public RFCs
```
go run ./dev/sg rfc list
....
....
....
RFC 747 WIP Role-Based Access Control
RFC 748 REVIEW Authorization for GraphQL API
RFC 750: Sourcegraph operator authentication provider for Cloud
```

## Test plan
Tested locally

### CODE REVIEW

File: dev/sg/sg_rfc.go [hunk 1]
```diff
@@ -12,17 +12,27 @@ var rfcCommand = &cli.Command{
...
+# List all Public RFCs
 sg rfc list
 
-# Search for an RFC
-sg rfc search "search terms"
+# List all Private RFCs
+sg rfc --private list
+
+# Search for an Public RFC
+sg rfc --private search "search terms"
```
Comment (burmudar) [inline dev/sg/sg_rfc.go:22]:
```suggestion
sg rfc search "search terms"

# Search for a Private RFC
sg rfc --private search "search terms"
```

File: dev/sg/sg_rfc.go [hunk 2]
```diff
@@ -12,17 +12,27 @@ var rfcCommand = &cli.Command{
...
 
 # Open a specific RFC
 sg rfc open 420
```
Comment (burmudar) [inline dev/sg/sg_rfc.go:25]:
```suggestion
sg rfc open 420

# Open a specific private RFC
sg rfc --private open 420
```

**Review by davejrt — APPROVED** [review]

**Comment by davejrt** [comment]
Just to clarify, we can search all (private and public) and delivery both in the same list because of RFC ID conflicts?

**Comment by BolajiOlajide** [comment]
> I'm guessing you meant "can't" because then you're right. We can technically merge the list but then when you do `sg rfc open` it will open the _first_ one which might be the private one, even if you meant the public one due to RFC number conflicts.

@eseliger started hacking a `sg rfc create <title>` command to make the ID conflicts reduced, this will potentially ensure all RFCs are numbered appropriately and also ensure they have the right title format.
You can take a look at the [draft PR here](https://github.com/sourcegraph/sourcegraph/pull/42757).

**Pull request merged**